src/Orleans/Storage/GenericRelationalStorage.cs

```diff
@@ -0,0 +1,210 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+using System.Threading.Tasks;
+using System.Linq;
+using System.Diagnostics;
+
+
+namespace Orleans.Runtime.Storage.Relational
+{
+    /// <summary>
+    /// A general purpose class to work with a given relational database.
+    /// </summary>    
+    [DebuggerDisplay("InvariantName = {InvariantName}, ConnectionString = {ConnectionString}")]
+    public class GenericRelationalStorage: IRelationalStorage
```

Comment:
Generic does not really feel like the right word because of the overloading with concepts of Generic<T>.

Looking at the usage patterns, I think you could just drop the prefix completely and call it simply `RelationalStorage`.